src/jimgw/jim.py

```diff
@@ -104,19 +104,30 @@ def posterior(self, params: Float[Array, " n_dim"], data: dict):
 
     def sample(self, key: PRNGKeyArray, initial_position: Array = jnp.array([])):
         if initial_position.size == 0:
-            initial_position = jnp.zeros((self.sampler.n_chains, self.prior.n_dim)) + jnp.nan
-        
-            while not jax.tree.reduce(jnp.logical_and, jax.tree.map(lambda x: jnp.isfinite(x), initial_position)).all():
-                non_finite_index = jnp.any(~jax.tree.reduce(jnp.logical_and, jax.tree.map(lambda x: jnp.isfinite(x), initial_position)),axis=1)
+            initial_position = (
+                jnp.zeros((self.sampler.n_chains, self.prior.n_dim)) + jnp.nan
+            )
+
+            while not jax.tree.reduce(
+                jnp.logical_and,
+                jax.tree.map(lambda x: jnp.isfinite(x), initial_position),
+            ).all():
+                non_finite_index = jnp.any(
+                    ~jax.tree.reduce(
+                        jnp.logical_and,
+                        jax.tree.map(lambda x: jnp.isfinite(x), initial_position),
+                    ),
+                    axis=1,
+                )
 
                 key, subkey = jax.random.split(key)
-                guess = self.prior.sample(subkey, self.sampler.n_chains)
+                guess = self.prior.sample(subkey, non_finite_index.sum())
```

Comment:
The guesses are not guaranteed to be finite, hence the original implementation. Please change it back